### PR TITLE
NAS-118819 / 22.12 / Handle edge case of malformed helm secrets

### PR DESCRIPTION
**Pull request by Qubad786**
## Problem

For a user we saw that his helm secret had become malformed due to reasons unknown which resulted in us failing to list any apps as when we queried secrets and tried to parse them to get their details that failed resulting in us failing to list down any apps.

## Solution

I explored helm behaviour in this case and noticed that `helm history` ignored the revision secret which was affected and showed down other valid revisions which were still available without having their secrets malformed.
Keeping in line with helm behaviour changes have been made to ignore decode exceptions in this case and just log that this happened once.

### CODE REVIEW

File: src/middlewared/middlewared/plugins/chart_releases_linux/secrets_management.py [hunk 1]
```diff
@@ -14,6 +15,8 @@
 
 class ChartReleaseService(Service):
 
+    LOGGED_SECRET = set()
```
Comment (yocalebo) [inline src/middlewared/middlewared/plugins/chart_releases_linux/secrets_management.py:18]: How large can this grow? I don't like idea of storing secret keys in memory for life of process while it's running.

Comment (Qubad786) [reply src/middlewared/middlewared/plugins/chart_releases_linux/secrets_management.py:18]: For most users it does not happen and for the user in question it was 2 secrets..
I don't expect it to be a widespread problem

**Review by sonicaj — APPROVED** [review]
Interesting case :P

File: src/middlewared/middlewared/plugins/chart_releases_linux/secrets_management.py (hunk 1 above)
Comment (yocalebo) [reply src/middlewared/middlewared/plugins/chart_releases_linux/secrets_management.py:18]: Okay, I trust you 😄

**Review by yocalebo — APPROVED** [review]

**Pull request merged**